### DISCUSSION TITLE
XPRESS is not generating the optimization log in the console

### DISCUSSION THREAD

**Discussion by davidhdezf** [google/or-tools#3002] [Linear Solver questions]
**What version of OR-tools and what language are you using?**
Version: source V7.5 stable
**Which solver are you using (XPRESS)**

**What operating system and version?**
Windows 10

**What did you do?**
Try to use xsolver.EnableOutput

**What did you expect to see*
Optimization Log in the console output

**What did you see instead?**
The log was not generated

****Problem and solution
In xpress_interface.cc the next code is used to generate the output 

  `  XPRSsetintcontrol(mLp, XPRS_OUTPUTLOG, 1); `

Nevertheless this code is not enough to generate the output the next code needs to be added. Note: I am printing all the messages levels.

```cpp
void XPRS_CC optimizermsg(XPRSprob prob, void* data, const char* sMsg, int nLen, int nMsgLvl);

void XPRS_CC optimizermsg(XPRSprob prob, void* data, const char* sMsg, int nLen, int nMsgLvl)
{
    switch (nMsgLvl) {

        /* Print Optimizer error messages and warnings */
    case 4:       /* error */
        printf("%*s\n", nLen, sMsg);
        break;
    case 3:       /* warning */
        printf("%*s\n", nLen, sMsg);
        break;

        /* Ignore other messages */
    case 2:       /* dialogue */
        printf("%*s\n", nLen, sMsg);
        break;
    case 1:       /* information */
        printf("%*s\n", nLen, sMsg);
        break;
   
        /* Exit and flush buffers */
    default:
        fflush(NULL);
        break;
    }
}
```

Next code should be added inside SOLVE before optimizing. 

`  /* Tell Optimizer to call optimizermsg whenever a message is output */
  XPRSsetcbmessage(mLp, optimizermsg, NULL);`

**Comment by sergun** [comment]
FYI: confirm solution of @davidhdezf works for me
